rfcs/0011-removal-managed-zone.md

```diff
@@ -0,0 +1,107 @@
+# Remove Managed Zone
+
+- Feature Name: `remove-managed-zone`
+- Start Date: 2024-06-26
+- RFC PR: [https://github.com/Kuadrant/architecture/pull/93](https://github.com/Kuadrant/architecture/pull/93)
+- Issue tracking: [https://github.com/Kuadrant/dns-operator/issues/166](https://github.com/Kuadrant/dns-operator/issues/166)
+
+# Summary
+[summary]: #summary
+
+ManagedZone was designed in the context of multiple tenants where the DNS operator was offered as a service and the owner of the service would own the credentials, the dns provider accounts and the zones being worked on. It had a requirement of adding and removing zones in the provider on demand. This is no longer the case, but the `ManagedZone` API has remained in place but is now mainly a reference to a credential for the underlying cloud DNS provider. We do not need this indirection and there is little value to the end user. So lets remove the ManagedZone API and change the DNSPolicy and DNSRecord to a localReference that defaults to a credential secret but is flexible enough to be expanded upon in the future.
+
+# Motivation
+[motivation]: #motivation
+
+The motiviation for these changes is to simplify the user experience of setting up a DNSPolicy by removing the need for an existing  `ManagedZone` API. As the current reference to the credential is in the ManagedZone API it makes sense to remove that API and move the reference to DNSPolicy.
+
+The key user impact here is to avoid users having to create additional none policy `Kuadrant APIs` in order to be able to leverage the `DNSPolicy` API. 
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+There will no longer be a `ManagedZone` resource that needs to be created. It will be expected that the zone already exists in the provider of choice. If it does not exist, you will see this reflected as an error state in the DNSPOlicy status. Credentials will continue to be stored in secrets by default, but these secrets will also be where any additional zone configuration is also applied such as the zone id to target and the root domain in that zone to use. To use different root domains is not currently possible and is outside of the scope of the RFC. 
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+change the DNSPolicy to have local object reference array for provider secrets in the same namespace.
+
+```
+spec:
+  providerRefs
+    - name: myawsroute53
```

Comment:
This is mentioned below. The DNS Operator will report an error in the existing sychronised status condition and this will be mirrored back into the DNSPolicy existing `recordConditions`